needrestart_info.py

```diff
@@ -0,0 +1,160 @@
+            elif key == "NEEDRESTART-KSTA":
+                self.kernel_status = KernelStatus(int(value))
+            # Microcode informations
+            elif key == "NEEDRESTART-UCCUR":
+                self.microcode_current_version = value
+            elif key == "NEEDRESTART-UCEXP":
+                self.microcode_expected_version = value
+            elif key == "NEEDRESTART-UCSTA":
+                self.microcode_status = MicroCodeStatus(int(value))
+            # Count the others
+            else:
+                needrestart_counter.update({key})
+
+        self.services_count = needrestart_counter["NEEDRESTART-SVC"]
+        self.containers_count = needrestart_counter["NEEDRESTART-CONT"]
+        self.sessions_count = needrestart_counter["NEEDRESTART-SESS"]
+
+
+def _write_timestamp(registry, needrestart_data):
+    g = Gauge(
+        "needrestart_timestamp",
+        "information about the version and when it was last run",
+        labelnames=["version"],
+        registry=registry,
+    )
+    g.labels(needrestart_data.version).set(needrestart_data.timestamp)
+
+
+def _write_kernel(registry, needrestart_data):
+    if needrestart_data.kernel_status:
+        e = Gauge(
+            "needrestart_kernel_status",
+            "information about the kernel status",
+            labelnames=["current", "expected"],
+            registry=registry,
+        )
+        e.labels(
+            needrestart_data.kernel_current_version,
+            needrestart_data.kernel_expected_version,
+        ).set(needrestart_data.kernel_status.value)
+
+
+def _write_microcode(registry, needrestart_data):
+    if needrestart_data.microcode_status:
+        e = Gauge(
+            "needrestart_microcode_status",
+            "information about the microcode status",
+            labelnames=["current", "expected"],
+            registry=registry,
+        )
+        e.labels(
+            needrestart_data.microcode_current_version,
+            needrestart_data.microcode_expected_version,
+        ).set(needrestart_data.microcode_status.value)
+
+
+def _write_services(registry, needrestart_data):
+    g = Gauge(
+        "needrestart_services_count",
+        "number of services requiring a restart",
+        registry=registry,
+    )
+    g.set(needrestart_data.services_count)
+
+
+def _write_containers(registry, needrestart_data):
+    g = Gauge(
+        "needrestart_containers_count",
+        "number of containers requiring a restart",
+        registry=registry,
+    )
+    g.set(needrestart_data.containers_count)
+
+
+def _write_sessions(registry, needrestart_data):
+    g = Gauge(
+        "needrestart_sessions_count",
+        "number of sessions requiring a restart",
+        registry=registry,
+    )
+    g.set(needrestart_data.sessions_count)
+
+
+def _main():
+    registry = CollectorRegistry()
+    needrestart_data = NeedrestartParser(
+        subprocess.run(["needrestart", "-b"], stdout=subprocess.PIPE)
```

Comment:
```suggestion
    needrestart_output = subprocess.run(["needrestart", "-b"], capture_output=True, text=True).stdout
    needrestart_data = NeedrestartParser(needrestart_output)
```

i think that makes it clearer that the process' output is to be used, the `text` option makes is unnecessary to decode it in the parsing logic.

the extra statement for capturing the output makes it better comprehensible imo.